linkerd/http-route/src/http/filter/redirect.rs

```diff
@@ -193,6 +193,84 @@ impl RedirectRequest {
     }
 }
 
+#[cfg(feature = "proto")]
+pub mod proto {
+    use super::*;
+    use linkerd2_proxy_api::{http_route as api, http_types};
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum InvalidRequestRedirect {
+        #[error("invalid location scheme: {0}")]
+        Scheme(#[from] http_types::InvalidScheme),
+
+        #[error("invalid HTTP status code: {0}")]
+        Status(#[from] http::status::InvalidStatusCode),
+
+        #[error("invalid HTTP authority: {0}")]
+        Authority(#[from] http::uri::InvalidUri),
+
+        #[error("invalid HTTP status code: {0}")]
```

Comment:
should this error maybe indicate that the status was too big to fit in 16 bits, to differentiate it from the other "invalid HTTP status code" message?